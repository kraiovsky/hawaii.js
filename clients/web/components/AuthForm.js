import React from 'react'
import { Form, Field } from 'react-final-form'
import PropTypes from 'prop-types'
import { required, mustBeEmail, composeValidators } from '../utils/validators'

const AuthForm = ({ onSubmit }) => {
  return (
    <Form
      onSubmit={onSubmit}
      render={({ handleSubmit, form, submitting, pristine, values }) => (
        <form onSubmit={handleSubmit}>
          <Field name="email" validate={composeValidators(required, mustBeEmail)}>
            {({ input, meta }) => (
              <div>
                <label htmlFor="email">Email</label>
                <input {...input} type="text" placeholder="example@email.com" id="email" />
                {meta.error && meta.touched && <span>{meta.error}</span>}
              </div>
            )}
          </Field>
          <div className="buttons">
            <button type="submit" disabled={submitting}>
              Submit
            </button>
            {/* eslint-disable-next-line react/jsx-handler-names */}
            <button type="button" onClick={form.reset} disabled={submitting || pristine}>
              Reset
            </button>
          </div>
        </form>
      )}
    />
  )
}

AuthForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
}

export default AuthForm
