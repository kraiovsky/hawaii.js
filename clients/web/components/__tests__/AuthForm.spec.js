import React from 'react'
import renderer from 'react-test-renderer'
import AuthForm from '../AuthForm'

const onSubmit = jest.fn()

test('Render AuthForm form', () => {
  const component = renderer.create(<AuthForm onSubmit={onSubmit} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
