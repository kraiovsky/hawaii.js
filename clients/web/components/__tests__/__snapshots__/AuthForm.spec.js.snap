// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render AuthForm form 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <label>
      Email
    </label>
    <input
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="example@email.com"
      type="text"
      value=""
    />
  </div>
  <div
    className="buttons"
  >
    <button
      disabled={false}
      type="submit"
    >
      Submit
    </button>
    <button
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Reset
    </button>
  </div>
</form>
`;
